src/test/rustdoc/mut-params.rs

```diff
@@ -0,0 +1,16 @@
+// Rustdoc shouldn't display `mut` in function arguments, which are
+// implementation details. Regression test for #81289.
+
+#![crate_name = "foo"]
+
+pub struct Foo;
+
+// @!has foo/struct.Foo.html '//*[@class="impl-items"]//*[@class="method"]' 'mut'
+impl Foo {
+    pub fn foo(mut self) {}
+
+    pub fn bar(mut bar: ()) {}
+}
+
+// @!has foo/fn.baz.html '//*[@class="rust fn"]' 'mut'
+pub fn baz(mut foo: Foo) {}
```

Comment:
I think I would rather have these `@!has` checks look like (replace `{path}` with the path):

```rust
// @!has {path} 'mut'
```

This way the tests won't silently stop testing if the class names or HTML structure change.

Reply:
This will break if any part of the HTML has the text 'mut', even if part of another word. Instead can you add it as two separate checks?
```
 // @has foo/fn.baz.html '//*[@class="rust fn"]'
 // @!has foo/fn.baz.html '//*[@class="rust fn"]' 'mut' 
```

Reply:
Yeah, that sounds good: it looks like the plain `@!has` is not working:

```
8: @!has check failed
 `PATTERN` did not match
 // @!has foo/struct.Foo.html 'mut'
15: @!has check failed
 `PATTERN` did not match
 // @!has foo/fn.baz.html 'mut'
Encountered 2 errors
```

Reply:
Actually, looking at `htmldocck`'s documentation, it seems like you'll need to do

```
@has foo/fn.baz.html '//*[@class="rust fn"]' {count}
```

where `{count}` is the number of times it occurs. Unless I'm reading the documentation wrong 🙃